### ISSUE TITLE
oidcSettings automaticSilentRenew false still results in silent renew

### ISSUE THREAD

**Issue by ckpearson** [perarnborg/vuex-oidc#71]
with `oidcSettings.js` looking as follows:

```
export const oidcSettings = {
  authority: 'authority',
  clientId: 'client-id',
  redirectUri: 'http://localhost:8080/oidc-callback',
  responseType: 'id_token token',
  scope: 'openid profile User.ReadBasic.All',
  silentRedirectUri: 'http://localhost:8080/oidc-silent-renew',
  automaticSilentRenew: false
}
```

I have a `beforeEnter` nav guard on my root route:

```
beforeEnter: (to, from, next) => {
      debugger
      let authed = lStore.getters['oidcStore/oidcIsAuthenticated']

      if (authed) {
        next('environments')
      } else {
        next()
      }
    }
```

And an oidc-silent-renew route as such:

```
  {
    path: '/oidc-silent-renew',
    meta: {
      isPublic: true
    },
    beforeEnter: (to, from, next) => {
      debugger
      vuexOidcProcessSilentSignInCallback(oidcSettings)
      next('/')
    }
  },
```

I'm seeing the nav guard for `/` being hit when first navigating to the app, and `authed` is false, but then inexplicably after a couple of seconds the `/oidc-silent-renew` root `beforeEnter` guard is getting hit with a new token.

I don't have any components in my hierarchy that are invoking any of the actions that kick off an oidc flow, so it appears as though `automaticSilentRenew` is being ignored.

**Comment by perarnborg** [comment]
This is the way I have design the package to work. If you specify a silentRedirectUri a silent sign in will be made on public routes if the user is not authenticated. The automaticSilentRenew setting on the other hand is what controls if a silent sign in should be made when tokens are about to expire.

If you do not want the silent signing to be performed you can remove the silentRedirectUri from your settings. Please let me know if this does not solve your problems.

**Comment by perarnborg** [comment]
Sorry, did not mean to close the issue. :)

**Comment by ckpearson** [comment]
That answers my question, I guess it was a case of confusion, as to my mind `automaticSilentRenew` being a bool seemed like it would be a toggle for *all* silent renewal, as both the public route and token expiry are "automatic".

Perhaps it might be an idea to have the behaviours both controllable via boolean toggles?

E.g.

`automaticSilentRenewExpiringTokens`
`automaticSilentRenewOnRoutes`

Naming is hard as always, but the current behaviour is fine now I know what it does. :+1:

**Comment by mediafreakch** [comment]
@perarnborg It seems this behaviour leads to a strange bug in my app. Where when I logout from a protected route I'm immediately being logged in again. As the `post_logout_redirect_uri` I configured a public route. I specified a `silent_redirect_uri` and `automaticSilentRenew: true` as well. 

I am confused about how to get this right...

**Comment by mediafreakch** [comment]
It might be worth to mention that in the Network tab, I see that the call to `/logout` (oidc provider) is being "canceled"...

**Comment by mediafreakch** [comment]
I found the problem. I was using vuetify with the following code: 

```
<v-list-item-action>
   <v-icon v-if="user" @click="signOut">mdi-logout</v-icon>
</v-list-item-action>
```

Turns out that the `signOut` (vuex-action) was called, but canceled somewhere up the tree because the click event propagated.

**Solution**

Attach `.stop` to the click handler like so:

```
<v-icon v-if="user" @click.stop="signOut">mdi-logout</v-icon>
```

With this I can actually keep `silent_redirect_uri` and `automaticSilentRenew` enabled and everything still behaves like I want.
